q2_feature_table/tests/test_summarize.py

```diff
@@ -210,10 +211,148 @@ def test_optional_inputs(self):
                 self.assertTrue('<td>3</td>' in file_text)
                 self.assertTrue('<td>4</td>' in file_text)
                 self.assertTrue('<td>6</td>' in file_text)
-                self.assertTrue('<td>1.0</td>' in file_text)
+                self.assertTrue('<td>a;b;e;d</td>' in file_text)
                 self.assertTrue('<td>a;b;c;d</td>' in file_text)
-                self.assertTrue('<td>0.4</td>' in file_text)
-                self.assertTrue('<td>0.7</td>' in file_text)
+                self.assertTrue('<td>08</td>' in file_text)
+                self.assertTrue('<td>17</td>' in file_text)
+
+    def test_optional_input_union(self):
+        seqs = DNAIterator(skbio.DNA(a, metadata=b)for a, b in (
+            ('A', {'id': 'seq01'}),
+            ('AA', {'id': 'seq02'}),
+            ('AAA', {'id': 'seq03'}),
+            ('AAAA', {'id': 'seq04'}),
+            ('AAAA', {'id': 'seq05'}),
+            ('AAA', {'id': 'seq06'}),
+            ('AA', {'id': 'seq07'}),
+            ('AAAAAAAAAA', {'id': 'seq08'})))
+
+        metadata = pd.DataFrame(index=['seq01', 'seq02',
+                                       'seq15', 'seq04',
+                                       'seq05', 'seq90',
+                                       'seq48', 'seq08'],
+                                columns=['att1', 'att2'],
+                                data=[['00', '01'], ['10', '11'],
+                                      ['03', '04'], ['12', '13'],
+                                      ['05', '06'], ['14', '15'],
+                                      ['07', '08'], ['16', '17']])
+        metadata.index.name = 'feature id'
+
+        taxonomy = pd.DataFrame([('a;b;c;d', '1.0'), ('a;b;c;f', '0.7'),
```

Comment:
Goes for all tests. Since we are accepting Collections of taxonomies now, probably split this into two taxonomies and put them in a dict then pass that dict into the test.